dataplane/src/packet_meta.rs

```diff
@@ -0,0 +1,125 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright Open Network Fabric Authors
+
+use routing::vrf::VrfId;
+use std::collections::HashMap;
+
+#[derive(Debug, Default)]
+pub struct InterfaceId(u32);
+#[allow(unused)]
+impl InterfaceId {
+    pub fn get_id(&self) -> u32 {
+        self.0
+    }
+    pub fn set_id(&mut self, id: u32) {
+        self.0 = id
+    }
+}
+
+#[derive(Debug, Default)]
+pub struct BridgeDomain(u32);
+#[allow(unused)]
+impl BridgeDomain {
+    pub fn get_id(&self) -> u32 {
+        self.0
+    }
+    pub fn with_id(id: u32) -> Self {
+        Self(id)
+    }
+}
+
+#[allow(unused)]
+#[derive(Copy, Clone, Debug, Eq, Hash, PartialEq)]
+pub enum DropReason {
+    InternalFailure,      /* catch-all for internal issues */
+    NotEthernet,          /* could not get eth header */
```

Comment:
I'm not sure `NotEthernet` is a valid branch.  As things stand, `Packet` doesn't even exist if it fails to parse even the eth header.

Reply:
... or are you saying that we would not get to see the packet at all ?